nucliadb/src/nucliadb/search/api/v1/find.py

```diff
@@ -126,6 +126,16 @@ async def find_knowledgebox(
     security_groups: list[str] = fastapi_query(SearchParamDefaults.security_groups),
     show_hidden: bool = fastapi_query(SearchParamDefaults.show_hidden),
     reranker: Reranker = fastapi_query(SearchParamDefaults.reranker),
+    user_metadata: Optional[str] = Query(
+        default=None,
+        title="User metadata",
+        description=(
```

Comment:
I would also put some validation to make sure the metadata does not exceed certain size. Say... 100KB ?

Reply:
We decided to remove this from GET endpoints, so now audit_metadata will only be available in POST endpoints

Reply:
Even in a POST, we should still restrict it to a max size IMO!
As otherwise it's a way to store arbitrary data in the audit database

Reply:
In the model, there are other fields, such as those of type str. Wouldn't we encounter the same issue with these fields, since a str can have an arbitrary size?
Why do we need to limit the size for this field and not for the others?

Reply:
IMO all params in an API should have limits, even if they are very big. Limits help you to scale the service and also the user to know how to deal properly with the api.

There are some params that don't have limits, and they should, but it's out of the scope of this PR to add them -- don't worry.

IMO `audit_metadata` should be limited in size (100kB or 1MB if you want), as it is arbitrary metadata that will be stored in the audit storage. This can help you reason about the right sizing of the persistant layers the audit service may need.

Reply:
Limits of 10KB added, I think 10KB is enough, we can always raise them later